Error accessing option type variables using `?`

### Describe the bug

Error accessing option type variables using `?`, whereas in functions returing options it works as expected.

```v 
a := ?int(none) // ?int(3)
y := a?
println(y)
```

### Expected Behavior

to get 3 for `?int(3)`

or `panic: option not set` for `none` which is the error message we get in case of functions returning `Option(none)`.

### Current Behavior

builder error for both `?int(none)` and `?int(3)`

```
==================
C:/Users/dhrit/AppData/Local/Temp/v_0/option1.15974423340005818176.tmp.c:7586: warning: cast between pointer and integer of different size
C:/Users/dhrit/AppData/Local/Temp/v_0/option1.15974423340005818176.tmp.c:12453: error: identifier expected
...
==================
```

### Reproduction Steps

see above

### Possible Solution

_No response_

### Additional Information/Context

_No response_

### V version

0.3.3 638805b

### Environment details (OS name and version, etc.)

```
OS: windows, Microsoft Windows 11 Home Single Language v22621 64-bit
Processor: 8 cpus, 64bit, little endian,
CC version: Error: exec failed (CreateProcess) with code 2: The system cannot find the file specified.
 cmd: cc --version

getwd: C:\Users\dhrit\Desktop\ToReead\vtest
vmodules: C:\Users\dhrit\.vmodules
vroot: C:\BIN\v
vexe: C:\BIN\v\v.exe
vexe mtime: 2023-03-09 04:58:45
is vroot writable: true
is vmodules writable: true
V full version: V 0.3.3 2597efa.638805b

Git version: git version 2.35.1.windows.2
Git vroot status: weekly.2023.10-17-g638805be (3 commit(s) behind V master)
.git/config present: true
thirdparty/tcc status: thirdparty-windows-amd64 e90c2620
```

--- a/vlib/v/gen/c/assign.v
+++ b/vlib/v/gen/c/assign.v
@@ -16,9 +16,7 @@ fn (mut g Gen) expr_with_opt_or_block(expr ast.Expr, expr_typ ast.Type, var_expr
 		g.writeln(';')
 		g.writeln('if (${expr}.state != 0) {')
 		if expr is ast.Ident && (expr as ast.Ident).or_expr.kind == .propagate_option {
-			g.write('return ')
-			g.gen_option_error(g.cur_fn.return_type, expr)
-			g.writeln(';')
+			g.writeln('\tpanic_option_not_set(_SLIT("none"));')
 		} else {
 			g.inside_or_block = true
 			defer {

